core/image.cpp

```diff
@@ -2379,6 +2380,47 @@ void Image::normalmap_to_xy() {
 	convert(Image::FORMAT_LA8);
 }
 
+void Image::bumpmap_to_normalmap(float bump_scale) {
+	ERR_FAIL_COND(!_can_modify(format));
+	convert(Image::FORMAT_RF);
+
+	PoolVector<uint8_t> result_image; //rgba output
+	result_image.resize(width * height * 4);
+
+	{
+		PoolVector<uint8_t>::Read rp = data.read();
+		PoolVector<uint8_t>::Write wp = result_image.write();
+
+		unsigned char *write_ptr = wp.ptr();
+		float *read_ptr = (float *)rp.ptr();
+
+		for (int ty = 0; ty < height; ty++) {
+			int py = ty + 1;
+			if (py >= height) py -= height;
+
+			for (int tx = 0; tx < width; tx++) {
+				int px = tx + 1;
+				if (px >= width) px -= width;
+				float here = read_ptr[ty * width + tx];
+				float to_right = read_ptr[ty * width + px];
+				float above = read_ptr[py * width + tx];
+				Vector3 up = Vector3(0, 1, (here - above) * bump_scale);
+				Vector3 across = Vector3(1, 0, (to_right - here) * bump_scale);
+
+				Vector3 normal = across.cross(up);
+				normal.normalize();
+
+				write_ptr[((ty * width + tx) << 2) + 0] = (128.0 + normal.x * 127.0);
```

Comment:
This maps [-1,1] to [1,255] rather than [0,255]

Reply:
Good catch.   (127.5 + normal.x * 127.5)  is right, yes?

Reply:
Looks right to me, up to possible rounding issues